src/analyses/variable-sensitivity/variable_sensitivity_object_factory.h

```diff
@@ -0,0 +1,254 @@
+/*******************************************************************\
+
+ Module: analyses variable-sensitivity
+
+ Author: Owen Jones owen.jones@diffblue.com
+
+\*******************************************************************/
+#ifndef CPROVER_ANALYSES_VARIABLE_SENSITIVITY_VARIABLE_SENSITIVITY_OBJECT_FACTORY_H
+#define CPROVER_ANALYSES_VARIABLE_SENSITIVITY_VARIABLE_SENSITIVITY_OBJECT_FACTORY_H
+
+#include <analyses/variable-sensitivity/array_abstract_object.h>
+#include <analyses/variable-sensitivity/constant_abstract_value.h>
+#include <analyses/variable-sensitivity/constant_array_abstract_object.h>
+#include <analyses/variable-sensitivity/constant_pointer_abstract_object.h>
+#include <analyses/variable-sensitivity/context_abstract_object.h>
+#include <analyses/variable-sensitivity/data_dependency_context.h>
+#include <analyses/variable-sensitivity/full_struct_abstract_object.h>
+#include <analyses/variable-sensitivity/interval_abstract_value.h>
+#include <analyses/variable-sensitivity/pointer_abstract_object.h>
+#include <analyses/variable-sensitivity/struct_abstract_object.h>
+#include <analyses/variable-sensitivity/union_abstract_object.h>
+#include <analyses/variable-sensitivity/value_set_abstract_object.h>
+#include <analyses/variable-sensitivity/write_location_context.h>
+#include <util/namespace.h>
+#include <util/options.h>
+
+struct vsd_configt
+{
+  struct
+  {
+    bool struct_sensitivity;
+    bool array_sensitivity;
+    bool pointer_sensitivity;
+  } primitive_sensitivity;
+
+  struct
+  {
+    bool data_dependency_context;
+    bool last_write_context;
+  } context_tracking;
+
+  struct
+  {
+    bool intervals;
+    bool value_set;
+  } advanced_sensitivities;
+
+  static vsd_configt from_options(const optionst &options)
+  {
+    vsd_configt config{};
+
+    if(
+      options.get_bool_option("value-set") &&
+      options.get_bool_option("data-dependencies"))
+    {
+      throw invalid_command_line_argument_exceptiont{
+        "Value set is not currently supported with data dependency analysis",
+        "--value-set --data-dependencies",
+        "--data-dependencies"};
+    }
+
+    config.primitive_sensitivity.struct_sensitivity =
+      options.get_bool_option("structs");
+    config.primitive_sensitivity.array_sensitivity =
+      options.get_bool_option("arrays");
+    config.primitive_sensitivity.pointer_sensitivity =
+      options.get_bool_option("pointers");
+
+    // This should always be on (for efficeny with 3-way merge)
+    // Does not work with value set
```

Comment:
❓ Do we know why it doesn't work?

Reply:
The comment is by @thk123 so I fear the answer to that may have left with him.  I don't know and regard it as a "to fix".